app/Traits/ClearsGlobalSearchCache.php

```
@@ -10,77 +10,119 @@ trait ClearsGlobalSearchCache
 
-        // For models with direct team_id property
-        if (property_exists($this, 'team_id') || isset($this->team_id)) {
-            return $this->team_id;
-        }
+            // For database models, team is accessed through environment.project.team
+            if (method_exists($this, 'team')) {
+                if ($this instanceof \App\Models\Server) {
+                    $team = $this->team;
+                } else {
+                    $team = $this->team();
+                }
```

PHPMD (2.15.0)

The method getTeamIdForCache uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)